scripts/commands/updateApiDocs.ts

```diff
@@ -148,7 +148,36 @@ zxMain(async () => {
   const outputDir = `${getRoot()}/docs/api/${pkg.name}`;
 
   console.log(`Deleting existing markdown for ${pkg.name}`);
-  await $`rm -rf ${outputDir}`;
+  await $`find ${outputDir}/* -not -path "*release-notes*" | xargs rm -rf {}`;
+
+  const legacyReleaseNoteVersions = (
+    await $`ls ${getRoot()}/docs/api/${pkg.name}/release-notes`.quiet()
```

Comment:
I'm surprised this works, given that the file is `legacy_release_notes`, right?

Also will this run on Provider and Runtime accidentally?

Finally, consider extracting this to a helper function since it's a lot imperative code that makes it harder to read the overall function flow.

Reply:
> I'm surprised this works, given that the file is `legacy_release_notes`, right?

Parsing `legacy_release_notes` into separate release note files was a one-time task (using the script in the PR description) since those notes are immutable. This PR extends `updateApiDocs.ts` to check for pre-existing release notes, and adds those notes to the TOC.

PR #352, extends `updateApiDocs.ts` to pull the release notes for the specified Qiskit version and add it to the `release-notes/` folder.

I'll try to explain this better in the PR and maybe add a comment too.

> Also will this run on Provider and Runtime accidentally?

Yes – fixed in e14ae29cf149b3764ab16c92c3dcbda6b9b5ca40

> ...consider extracting this to a helper function...

Good call, also fixed in e14ae29cf149b3764ab16c92c3dcbda6b9b5ca40